### PR TITLE
Namespace checks

### PR DESCRIPTION
**Pull request by charlienegri**
**Summary**:
if there is no env_string we still need to check that the namespace does not contain a .dev or .staging, as it could be as a result of a mistake so far not prevented at the point of ingestion

**Related issue**: 
#229

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: dmci/api/app.py [hunk 1]
```diff
@@ -235,7 +235,14 @@ def _check_metadata_id(metadata_id, env_string=None):
             return None, None, f"Cannot convert to UUID: {uuid_str}"
 
         # Check that the namespace is correct
-        if env_string is not None:
+        if not env_string:  # we are in prod
```
Comment (mortenwh) [inline dmci/api/app.py:238]: Better:
```suggestion
        if env_string is None:  # we are in prod
```

**Review by mortenwh — CHANGES_REQUESTED** [review]
One minor change, and github reports a conflict.

File: dmci/api/app.py (hunk 1 above)
Comment (charlienegri) [reply dmci/api/app.py:275]: this broke the tests, I reverted

File: dmci/api/app.py [hunk 2]
```diff
@@ -272,7 +272,14 @@ def _check_metadata_id(metadata_id, env_string=None):
...
```
[the remark this answers is not in the document]
Comment (charlienegri) [reply dmci/api/app.py:275]: sure I'll change the test

**Review by mortenwh — APPROVED** [review]

**Pull request merged**